database/database.go

```diff
@@ -193,79 +189,25 @@ func (db *DB) UserDelete(ctx context.Context, u *User) error {
 	return nil
 }
 
-// UserUpdate saves the user in the DB.
+// UserUpdate changes the user's data int he DB.
+// It never changes the id or sub of the user.
 func (db *DB) UserUpdate(ctx context.Context, u *User) error {
-	if !u.Email.Validate() {
-		return ErrInvalidEmail
-	}
-	// Check for an existing user with this email.
-	users, err := db.managedUsersByField(ctx, "email", string(u.Email))
-	if err != nil && !errors.Contains(err, ErrUserNotFound) {
-		return errors.AddContext(err, "failed to query DB")
-	}
-	// Sanity check.
-	if len(users) > 1 {
-		build.Critical("More than one user found with email", u.Email)
-		return ErrEmailAlreadyUsed
-	}
-	if len(users) > 0 && !bytes.Equal(u.ID[:], users[0].ID[:]) {
-		return ErrEmailAlreadyUsed
-	}
-	// TODO What if we have a race a user gets this email right at this point?
 	// Update the user.
 	filter := bson.M{"_id": u.ID}
 	update := bson.M{"$set": bson.M{
-		"_id":       u.ID,
-		"firstName": u.FirstName,
-		"lastName":  u.LastName,
-		"email":     u.Email,
+		"_id":  u.ID,
```

Comment:
I'm surprised mongo this works, it's not something you'd usually do. A document's `_id` is never overwritten. it's redundant so I would definitely remove it here.